R/writer.R

```diff
@@ -116,20 +116,45 @@ write_errors <- function() {
 #' @return A formatted vector of warnings
 #' @export
 #'
-#' @examples
-#' scriptPath <- tempfile()
-#' logDir <- tempdir()
-#' writeLines("print('hello timber')", con = scriptPath)
+write_warnings <- function() {
+   warnings <- get_log_element("warnings")
+
+   paste0("\nWarnings:\n\t",
+          paste0(warnings, collapse = "\n\t"))
+}
+
+#' Format messages attribute for writing
 #'
-#' log_remove()
+#' @return A formatted vector of messages
+#' @export
 #'
-#' log_config(scriptPath)
+write_messages <- function() {
+   messages <- get_log_element("messages")
+
+   paste0("Messages:\n\t",
+          paste0(messages, collapse = "\n\t"))
+}
+
+#' Format output attribute for writing
 #'
-#' write_warnings()
+#' @return A formatted vector of output
+#' @export
 #'
-write_warnings <- function() {
-   warnings <- get_log_element("warnings")
+write_output <- function() {
+   output <- get_log_element("output")
 
-   paste0("Warnings:\n\t",
-          paste0(warnings, collapse = "\n\t"))
+   paste0("Output:\n\t",
+          paste0(output, collapse = "\n\t"))
+}
+
+#' Format result attribute for writing
+#'
+#' @return A formatted vector of results
+#' @export
+#'
+write_result <- function() {
+   result <- get_log_element("result")
+
+   paste0("\nResult:\n",
+          paste0(result, collapse = "\n\t"))
```

Comment:
This can get tricky since result could be anything; dataframe, model object, etc.  What do you think about letting print for each object handle how we present it in the log?

```suggestion
capture.output(print(result$value))
```